app/models/plugin.rb

```diff
@@ -11,6 +11,11 @@ class Plugin < ActiveRecord::Base
   delegate :url,   to: :approved_script, allow_nil: true
   delegate :version, to: :approved_script, allow_nil: true
 
+  after_initialize :generate_rare_key
```

Comment:
Is it desired that we automatically set plugins to have shared state? I don't really know, just thinking loudly.
If we always generate UUID (that can be overwritten in case of need), we could have one lookup strategy less - `PluginAndRun` seems redundant in this case. Unique `shared_learner_state_key` would handle it.